### PR TITLE
Variable label placement example

### PR DESCRIPTION
**Pull request by osana**
This PR adds an example of variable label placement using [PropertyFactory.textVariableAnchor().](https://github.com/mapbox/mapbox-gl-native/blob/20a807fdf36e2609973451ef8d0e49472a59db91/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/layers/PropertyFactory.java#L2324)

The renderer attempts placing the label at the positions in this list in the provided order of the array at collision detection time. If the symbol was successfully placed in a previous round, the renderer attempts that anchor position first before moving on to the rest of the list.

In this example the following values were used:
```
textVariableAnchor: ["top", "bottom", "left", "right"],
textRadialOffse: 0.5
textJustify: auto
```

ezgif com-optimize

### CODE REVIEW

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/VariableLabelPlacementActivity.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                        textColor(Color.RED),
+                        textVariableAnchor(
+                            new String[]{TEXT_ANCHOR_TOP, TEXT_ANCHOR_BOTTOM, TEXT_ANCHOR_LEFT, TEXT_ANCHOR_RIGHT}),
+                        textJustify(TEXT_JUSTIFY_AUTO),
+                        textRadialOffset(0.5f))),
+            new Style.OnStyleLoaded() {
+              @Override
+              public void onStyleLoaded(@NonNull final Style style) {
+                VariableLabelPlacementActivity.this.mapboxMap = mapboxMap;
+                new LoadGeoJson(style,VariableLabelPlacementActivity.this).execute();
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/VariableLabelPlacementActivity.java]: We are currently trying to load a local geojson file and aren't manipulating its contet. We can remove a lot of this boilerplate code if we rely on the loading map resources through assets. Us  `asset://poi_places.geojson` as url for the second parameter of the geojsonsource constructor. 

The json loader can be removed and the whole asynctask setup

Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/VariableLabelPlacementActivity.java]: Got rid of the `AsyncTask` and `onDataLoaded()`.

Doing this didn't work for me (lol):
Screen Shot 2019-05-22 at 4 51 53 PM

So, I used the `loadJsonFromAsset()` method that's used in some other examples:

https://github.com/mapbox/mapbox-android-demo/blob/fff1abe14114b4d7691996d1cba5be548c305e5d/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/VariableLabelPlacementActivity.java#L71
https://github.com/mapbox/mapbox-android-demo/blob/fff1abe14114b4d7691996d1cba5be548c305e5d/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/VariableLabelPlacementActivity.java#L95-L109

Example's working the same as seen in the GIF in OP.

Did I misinterpret your instructions? Is there some even smoother way to pull from the assets folder?

Comment (tobrun) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/VariableLabelPlacementActivity.java]: My bad, didn't double check the constructors. The issue with this setup is that you are trying to load that URI as an geojson string. 
I tried using the URL constructor but that failed as that URI is not a valid URL. 
Was able to make it work as:

```java
    source = new GeoJsonSource(MARKER_SOURCE);
    source.setUrl("asset://something.geojson");
```

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/utils/JsonLoader.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import timber.log.Timber;
+
+/**
+ * This class is a helper class to asynchrously load data
+ * while holding a weak reference to a callback.
+ * T data that is loaded
+ * @param <T> data to be loaded
+ */
+public class JsonLoader<T> extends AsyncTask<Void, Void, T> {
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/utils/JsonLoader.java]: class can be removed

Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/utils/JsonLoader.java]: Gah, keep forgetting to delete that damn unused class. Still seeing weird behavior, so it's still WIP

ezgif com-resize (1)

Comment (tobrun) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/utils/JsonLoader.java]: that issue is https://github.com/mapbox/mapbox-gl-native/issues/14715, feel free to ignore this bug

Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/utils/JsonLoader.java]: Awesome 🚀

**Pull request merged**